### PR TITLE
🔒 fix(cmux): harden CMUX_CLI resolution against CWE-426/427

### PR DESCRIPTION
**Pull request by takeuchi-shogo**
## Summary
- `CMUX_CLI` 解決の PATH hijacking (CWE-426) と symlink hijack (CWE-427) を mitigation
- 共通 lib `scripts/lib/cmux_resolver.sh` に集約 (3 sh で source、cmux-notify.sh は path 構造の都合で inline 維持)
- 関連: #47 で導入された `command -v` fallback が Gemini HIGH (CWE-426) として指摘された経緯への対応

## Changes
1. **env validation**: `$CMUX_CLI` を絶対パス + 実行可能 + 非ディレクトリ + 非 symlink で検証、不正なら stderr + return 1
2. **fail-closed**: `command -v` fallback を完全削除、PATH 検索なし
3. **allow-list**: macOS DMG インストールの `/Applications/cmux.app/Contents/Resources/bin/cmux` 1 件に絞る (Homebrew/Nix/asdf 等は symlink 利用が標準のため env override で対応)
4. **CWE-427 対策**: env / allow-list 両方で symlink reject

## Scope
- **防御対象**: PATH 上の任意 dir からの hijack (CWE-426)、symlink 経由 hijack (CWE-427)
- **非対象**: user uid compromise (allow-list path の user 所有を許容)、validation → execve 間の TOCTOU race window

## Review process
2 ラウンドの multi-reviewer code review を実施 (codex-reviewer / security-reviewer / gemini-explore / code-reviewer / cross-file-reviewer / edge-case-hunter)。Round 1 で HIGH 3 件、Round 2 で Critical 1 件 (allow-list 内 Homebrew パスが symlink reject で機能不能の矛盾) を発見・解消。

## Test plan
- [x] \`bash -n\` 5/5 PASS
- [x] resolver 動作検証 6 シナリオ PASS (env valid/reject 4 種、allow-list hit、Homebrew via env canonical)
- [x] cmux-notify.sh fallback 経路: env reject → CMUX_CLI="" → osascript fallback 確認
- [x] lefthook pre-commit / commit-msg 全通過

### CODE REVIEW

File: scripts/lib/cmux_resolver.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    fi
+
+    local c
+    for c in \
+        "/Applications/cmux.app/Contents/Resources/bin/cmux"; do
+        if [[ -x "$c" ]] && [[ ! -d "$c" ]] && [[ ! -L "$c" ]]; then
+            printf '%s\n' "$c"
+            return 0
+        fi
+    done
```
Comment (takeuchi-shogo) [inline scripts/lib/cmux_resolver.sh:28]: 🟢 nit: エントリが1件のみのため `for` ループは不要。`if` で直接書くと意図が明確になる。
将来エントリを追加する際に `for` に戻せばよい。
```suggestion
    local c="/Applications/cmux.app/Contents/Resources/bin/cmux"
    if [[ -x "$c" ]] && [[ ! -d "$c" ]] && [[ ! -L "$c" ]]; then
        printf '%s\n' "$c"
        return 0
    fi
```

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/runtime/collect-result.sh [hunk 2]
```diff
@@ -12,8 +12,9 @@ set -euo pipefail
 SCRIPT_DIR="$(cd "$(dirname "$0")" && pwd)"
 LIB_DIR="$(cd "$SCRIPT_DIR/../lib" 2>/dev/null && pwd || echo "$SCRIPT_DIR")"
 source "${LIB_DIR}/dispatch_logger.sh"
+source "${LIB_DIR}/cmux_resolver.sh"
 
-CMUX_CLI="/Applications/cmux.app/Contents/Resources/bin/cmux"
+CMUX_CLI="$(_resolve_cmux_cli)" || { echo "[collect-result] cmux not found" >&2; exit 3; }
```
Comment (takeuchi-shogo) [inline scripts/runtime/collect-result.sh:17]: 💭 question: `exit 3` はファイルヘッダ (`# Exit: 0=completed, 1=error, 2=timeout`) に未記載。
`launch-worker.sh` / `race-runner.sh` も同様。ヘッダに `3=cmux-not-found` を追記するか、`exit 1` に統一すると揃う。

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by takeuchi-shogo — COMMENTED** [review]
🤖 自動レビュー: 🔴 0 / 🟡 0 / 🟢 1 / 💭 1 (詳細はインライン)
🔧 自動修正: none (shfmt/nixfmt/shellcheck 未インストール・typo なし、bash -n 5/5 PASS)
---
_Generated by Claude Code Routine_

---
_Generated by [Claude Code](https://claude.ai/code)_
